tutorials/getstarted.qmd

```diff
@@ -1,3 +1,312 @@
+A Lie group is usually written as a tuple $(ℳ,∘)$ os the manifold and the group operation thereon.
+
+For more theoretical background, see for example [HilgertNeeb:2012](@cite), especially their Chapter 9.
+
+## Rotations on the plane $ℝ^2$ and in $ℝ^3$.
+
+This first part considers rotations in the plane. These can be represented by rotation matrices
+```math
+R_α = \begin{pmatrix} \cos(α),& -\sin(α)\\ \sin(α) & \cos(α) \end{pmatrix},\qquad α ∈ ℝ,
+```
+
+where we already represent all possible rotations when restricting $α$ to $[0,2π]$.
+Adding two matrices $R_α + R_β$ does not yield a valid rotation matrix again, but we
+can have these matrices in mind as being “locally like the real line”, since changing the rotation angle
+slightly yields rotation matrices that are “close by” in the sense that they perform nearly the same rotation
+when applied to a point $x ∈ ℝ^2$ by computing $R_αx$.
+Matrix multiplication $R_β = R_{α_1}*R_{α_1}$ does yield a new rotation and also fulfils all other
+properties to yield a group operation. Using trigonometric identities we observe that $β = α_1 + α_2$.
+
+The set of rotation matrices is hence a Lie group and called the [special orthogonal group](../groups/special_orthogonal_group.md) $\mathrm{SO}(2)$. In `LieGroups.jl` we initialise this as
+
+```{julia}
+using LieGroups, LinearAlgebra, RecursiveArrayTools
+SO2 = SpecialOrthogonalGroup(2)
+```
+
+Elements $g, h ∈ \mathrm{SO}(2)$ we generate as
+
+```{julia}
+g = [1.0 0.0; 0.0 1.0]
+h = 1/sqrt(2) .* [1.0 -1.0; 1.0 1.0]
+```
+
+A first thing to do is to use [`is_point`](@ref) to verify they are valid.
+
+```{julia}
+(is_point(SO2, g), is_point(SO2, h))
+```
+
+The already mentioned group operation as the matrix multiplication, there is also a generic
+function available, [`compose`](@ref) as well as its in-place variant [`compose!`](@ref).
+The following two yield the same matrix
+
+```{julia}
+[ compose(SO2, g, h), g*h ]
+```
+
+Furthermore a lot of of functions are “passed down” tp the manifold, which is stored within
+the [`LieGroup`](@ref). For example the dimension of the manifold, or the number of degrees of freedom
+can be accessed via the ``[`manifold_dimension`](@extref `ManifoldsBase.manifold_dimension-Tuple{AbstractManifold}`)``{=commonmark}
+
+```{julia}
+manifold_dimension(SO2)
+```
+
+## The Lie algebra, and the Lie group exponential function.
+
+For the following investigations, we consider the special orthogonal group $\mathrm{SO}(3)$,
+that is rotations in a 3-dimensional space.
+Besides one special rotation in the following code for completeness, a prominent, at first glace
+maybe a bit “dull” point on this Lie group is the identity matrix `e`.
+
+```{julia}
+SO3 = SpecialOrthogonalGroup(3)
+g = [cos(π/3) -sin(π/3) 0.0; sin(π/3) cos(π/3) 0.0; 0.0 0.0 1.0]* [cos(π/4) 0.0 -sin(π/4); 0.0 1.0 0.0; sin(π/4) 0.0 cos(π/4)]
```

Comment:
Maybe it would be more clear to use Rotations.jl here? You could just do something like `g = RotZ(π/3) * RotY(π/4)`.

Reply:
Sure, fair enough. Makes the code more readable. Will do that.